### PR TITLE
New Concrete API

### PR DESCRIPTION
**Pull request by henriquemoody**
## Motivation

- We don't have enough granularity control. Therefore rules like `not()` cannot work properly;
- `Validatable` interface has too many methods;
- The current API makes it hard to handle messages and translations;
- All rules depend on `AbstractRule`;.

***
Related to #354

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ before_script:
   - composer install --no-interaction --prefer-source
 
 script:
-  - vendor/bin/phpunit --configuration phpunit.xml.dist --colors --coverage-clover=coverage.clover
+  - vendor/bin/phpunit --testsuite=unit --configuration phpunit.xml.dist --colors --coverage-clover=coverage.clover
```
Comment (henriquemoody) [inline .travis.yml]: It must be removed before merge.

**Comment by prisis** [comment]
hey @henriquemoody, can you merge this in master or a v2 branch so we can help you?

**Comment by henriquemoody** [comment]
@prisis, I don't want to mess up with master now, but how about you send the pull requests to my fork?

**Comment by prisis** [comment]
@henriquemoody i will :)

**Comment by henriquemoody** [comment]
I'm closing this one since we'll be focusing on launching the 2.0 without these changes.

@prisis, if you still want to contribute, check out #921. 🐼

**Pull request closed without merging**